lib/ocpp/v16/charge_point_impl.cpp

```diff
@@ -890,16 +890,21 @@ std::optional<MeterValue> ChargePointImpl::get_latest_meter_value(int32_t connec
 
             case Measurand::Temperature: {
                 // temperature
-                const auto temperature = measurement.temperature_C;
-                if (temperature) {
+                for (const auto temperature : measurement.temperature_C) {
                     sample.unit.emplace(UnitOfMeasure::Celsius);
-                    if (temperature.value().location.has_value()) {
-                        sample.location.emplace(conversions::string_to_location(temperature.value().location.value()));
-                    } else {
-                        sample.location.emplace(Location::EV);
+                    if (temperature.location.has_value()) {
+                        try {
+                            sample.location.emplace(conversions::string_to_location(temperature.location.value()));
+                        } catch (const StringToEnumException& e) {
+                            EVLOG_warning << "Could not convert string: " << temperature.location.value()
+                                          << " to Location";
```

Comment:
Id rather set this to `EVLOG_debug` since it will repeately be printed if the location cant be converted for every meter value